src/api/mod.rs

```diff
@@ -334,6 +367,16 @@ impl<'a> ValueListBuilder<'a> {
         self
     }
 
+    pub fn metadata(mut self, key: &str, value: MetaValue) -> ValueListBuilder<'a> {
```

Comment:
Since `metadata` requires ownership of `key` it may be best to change the argument to a `String`. (ref [C-CALLER-CONTROL](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control))

Reply:
Looking into this a bit further, were you able to get a submit value list to work with a hashmap with borrowed keys?

```rust
struct SubmitValueList<'a> {
    // ...
    meta: Option<HashMap<&'a str, MetaValue>>,
}
```

It looks like this might be possible.

Reply:
yes it works ! i feared to have to handle strange borrow checker error but it all went good :)